src/SFML/Network/IpAddress.cpp

```diff
@@ -210,7 +225,16 @@ bool operator !=(const IpAddress& left, const IpAddress& right)
 ////////////////////////////////////////////////////////////
 bool operator <(const IpAddress& left, const IpAddress& right)
 {
-    return left.toInteger() < right.toInteger();
+    if (left.m_valid < right.m_valid)
+        return true;
+
+    if (right.m_valid < left.m_valid)
+        return false;
+
+    if (left.m_address < right.m_address)
+        return true;
+
+    return false;
 }
```

Comment:
> How to you easily extend it if you add a 3rd member, with C++03 code?

```cpp
return std::make_pair(std::make_pair(left.m_valid, left.m_address), left.m_third)
     < std::make_pair(std::make_pair(right.m_valid, right.m_address), right.m_third);
```
Now, how do you easily extend binary1248's code to a third member? :grimacing: 

> The operations are equivalent.

I know, but shouldn't we go for the shorter and easier-to-understand code if we have two options of writing it down? Or what's the deal with `std::pair`? :wink: